ibis/expr/scope.py

```diff
@@ -122,17 +116,17 @@ def merge_scope(self, other_scope: 'Scope', overwrite=False) -> 'Scope':
         """
         result = Scope()
 
-        for op in self.items():
-            result[op] = self[op]
+        for op in self._items.keys():
```

Comment:
Is this basically what you want?
https://stackoverflow.com/questions/6354436/python-dictionary-merge-by-updating-but-not-overwriting-if-value-exists

```
new_items = {}

if overwrite:
    new_items = dict(list(self._items.items()) + list(other._items.items()))
else:
    new_items = dict(list(other._items.items()) + list(self._items.items()))

return new Scope(new_items)
```

Reply:
This is nice but to update we cannot simply do these. We should think about time context and the logic is covered in `result.get_value(op, v.timecontext) is None`

Reply:
I see. Ok what you have is fine.

Reply:
why don't we define
```
def __iter__(self):
    return iter(self._items.keys())
```

then 
```
for op in self:
   ...
````

will just work

or is this more confusing?

Reply:
This is fine, scope is indeed iterable, I will do that